core/src/appleMain/kotlin/ScannerBuilder.kt

```diff
@@ -34,8 +35,8 @@ public actual class ScannerBuilder {
         allowDuplicateKeys?.also {
             options[CBCentralManagerScanOptionAllowDuplicatesKey] = it
         }
-        solicitedServiceUuids?.also {
-            options[CBCentralManagerScanOptionSolicitedServiceUUIDsKey] = it.toTypedArray()
+        solicitedServiceUuids?.also { uuids ->
+            options[CBCentralManagerScanOptionSolicitedServiceUUIDsKey] = uuids.map { it.toCBUUID() } as NSArray
```

Comment:
I don't think this cast is necessary and can be simplified to:

```suggestion
            options[CBCentralManagerScanOptionSolicitedServiceUUIDsKey] = uuids.map(Uuid::toCBUUID)
```

Reply:
I thought this would crash, but just ran with it and it looks good!